pkg/capture/capture_test.go

```diff
@@ -0,0 +1,96 @@
+package capture
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestLowTrafficDeadlock(t *testing.T) {
+	testDeadlock(t, 0)
+	testDeadlock(t, 1)
+	testDeadlock(t, 10)
+}
+
+func TestHighTrafficDeadlock(t *testing.T) {
+	testDeadlock(t, -1)
+}
+
+func testDeadlock(t *testing.T, maxPkts int) {
+	ctx := context.Background()
+	mockSrc := newMockCaptureSource(maxPkts)
+	mockC := &Capture{
```

Comment:
What about a `newCapture()` utility function?

Reply:
This will be completely replaced by the mock sources from `slimcap` in #88 , thereby making all this redundant...